lib/metasploit/framework/ssh/platform.rb

```diff
@@ -0,0 +1,121 @@
+module Metasploit
+  module Framework
+    module Ssh
+      module Platform
+        def self.get_platform(ssh_socket)
+          info = get_platform_info(ssh_socket, timeout: 10)
+          get_platform_from_info(info)
+        end
+
+        def self.get_platform_info(ssh_socket, timeout: 10)
```

Comment:
Probably want to pull out the actual platform info acquisition for relevant platforms to an appropriate mixin, and reference from here in terms of gluing together the SSH calls.
If we can get the meterssh style full channelized sessions into the mix, then this will be a bunch easier too.

Reply:
Moving it into a mixin under the MSF namespace would complicate it's usage in the existing SSH brute force code from which I moved it that exists under the Metasploit namespace. It was my understanding through observation that MSF could include Metasploit, but not the other way around just like with Rex.